lms/services/assignment.py

```diff
@@ -10,55 +12,110 @@ def __init__(self, db):
         self._db = db
 
     @lru_cache(maxsize=128)
-    def get(self, tool_consumer_instance_guid, resource_link_id):
-        return (
+    def get(
+        self,
+        tool_consumer_instance_guid,
+        resource_link_id=None,
+        ext_lti_assignment_id=None,
+    ):
+        """
+        Get an assignment using using either resource_link_id or ext_lti_assignment_id.
+
+        When a canvas assignment has just been configured it will have
+            resource_link_id=None and ext_lti_assignment_id!=None
+        until it's first lunched.
+        """
+        assignment = (
             self._db.query(Assignment)
-            .filter_by(
-                tool_consumer_instance_guid=tool_consumer_instance_guid,
-                resource_link_id=resource_link_id,
+            .filter(
+                Assignment.tool_consumer_instance_guid == tool_consumer_instance_guid,
+                (
+                    (Assignment.resource_link_id == resource_link_id)
+                    | (Assignment.ext_lti_assignment_id == ext_lti_assignment_id)
```

Comment:
Not sure if clear from the comment but this can't be an `and` there will be a time when we have stored the ext_lti_assignment_id but not the resource_link_id but we have both !=None in this function (on  first launch).

Reply:
And I guess we have pre-existing assignments that have a `resource_link_id` but no `ext_lti_assignment_id`. And also non-Canvas assignments too. So we do need to support looking up by `resource_link_id` only.

I'm wondering about this query and `None` / `NULL` though. For example if the call is `get(tool_consumer_instance_guid, resource_link_id, ext_lti_assignment_id=None)` won't that match *all* rows that have `NULL` in the `ext_lti_assignment_id`?